src/main/java/uk/gov/ons/ctp/response/sample/ingest/CsvIngesterBusiness.java

```diff
@@ -26,138 +32,197 @@
-    Set<String> unitRefs = new HashSet<>();
-
-      while((nextLine = csvReader.readNext()) != null) {
-        try {
-          sampleUnitList.add(parseLine(nextLine, unitRefs));
-        } catch(CTPException e){
-          String newMessage = String.format("Line %d: %s", csvReader.getRecordsRead(), e.getMessage());
-          throw new CTPException(e.getFault(), e, newMessage);
+    private static final String SAMPLEUNITREF = "sampleUnitRef";
+    private static final String FORMTYPE = "formType";
+    private static final String CHECKLETTER = "checkletter";
+    private static final String FROSIC92 = "frosic92";
+    private static final String RUSIC92 = "rusic92";
+    private static final String FROSIC2007 = "frosic2007";
+    private static final String RUSIC2007 = "rusic2007";
+    private static final String FROEMPMENT = "froempment";
+    private static final String FROTOVER = "frotover";
+    private static final String ENTREF = "entref";
+    private static final String LEGALSTATUS = "legalstatus";
+    private static final String ENTREPMKR = "entrepmkr";
+    private static final String REGION = "region";
+    private static final String BIRTHDATE = "birthdate";
+    private static final String ENTNAME1 = "entname1";
+    private static final String ENTNAME2 = "entname2";
+    private static final String ENTNAME3 = "entname3";
+    private static final String RUNAME1 = "runame1";
+    private static final String RUNAME2 = "runame2";
+    private static final String RUNAME3 = "runame3";
+    private static final String TRADSTYLE1 = "tradstyle1";
+    private static final String TRADSTYLE2 = "tradstyle2";
+    private static final String TRADSTYLE3 = "tradstyle3";
+    private static final String SELTYPE = "seltype";
+    private static final String INCLEXCL = "inclexcl";
+    private static final String CELLNO = "cell_no";
+    private static final String CURRENCY = "currency";
+
+    private static final String[] COLUMNS = new String[]{SAMPLEUNITREF, CHECKLETTER, FROSIC92, RUSIC92, FROSIC2007,
+            RUSIC2007, FROEMPMENT, FROTOVER, ENTREF, LEGALSTATUS, ENTREPMKR, REGION, BIRTHDATE, ENTNAME1, ENTNAME2, ENTNAME3,
+            RUNAME1, RUNAME2, RUNAME3, TRADSTYLE1, TRADSTYLE2, TRADSTYLE3, SELTYPE, INCLEXCL, CELLNO, FORMTYPE, CURRENCY};
+    @Autowired
+    private SampleService sampleService;
+    @Autowired
+    private AppConfig appConfig;
+    @Autowired
+    private PartyPublisher partyPublisher;
+
+    private ColumnPositionMappingStrategy<BusinessSampleUnit> columnPositionMappingStrategy;
+
+    public CsvIngesterBusiness() {
+        columnPositionMappingStrategy = new ColumnPositionMappingStrategy<>();
+        columnPositionMappingStrategy.setType(BusinessSampleUnit.class);
+        columnPositionMappingStrategy.setColumnMapping(COLUMNS);
+    }
+
+    /**
+     * Lazy create a reusable validator
+     *
+     * @return the cached validator
+     */
+    @Cacheable(cacheNames = "csvIngestValidator")
```

Comment:
I know you didn't add this but it's really strange to cache an object that could just be created on construction. Just had to point it out

Reply:
I have no idea about this - I kind of agree but I don't want to remove it because I don't know why it's there in the first place